Django/BookLibrary/library/views.py

```diff
@@ -0,0 +1,128 @@
+from django.shortcuts import render, redirect
+from rest_framework.views import APIView
+from rest_framework.permissions import AllowAny, IsAuthenticated
+from rest_framework.authentication import SessionAuthentication, BasicAuthentication
+
+from .forms import SearchBook
+from .models import Book, SearchHistory
+from .book_details import BookDetails
+from .serializers import BookSerializer
+from .recommendations import Recommendations
+
+'''Home Page View'''
+class Home(APIView):
+    permission_classes = [AllowAny]
+    authentication_classes = [SessionAuthentication, BasicAuthentication]
+    
+    def home(request):
+        if request.user.is_authenticated:
+            return render(request, 'library/home.html')
+        else:
+            return render(request, 'authentication/signin.html')
+        
+'''This class returns the Search result of the Book'''
+class SearchBooks(APIView):
+    permission_classes = [IsAuthenticated]
+    authentication_classes = [SessionAuthentication, BasicAuthentication]
+    
+    def get(self, request, format=None):
+        input_serializer = BookSerializer()
+        return render(request, 'library/base.html', {'form':input_serializer})
+
+    def post(self, request, format=None):
+        input_serializer = BookSerializer(data=request.data)
+        if input_serializer.is_valid():
+            title = input_serializer.validated_data['title'].title()
+            details = BookDetails(title)
+            amazon_url = details.amazon_link
+            flipkart_url = details.flipkart_link
+            good_reads_review = details.good_reads_review
+            cover_image = details.cover_image
+            # Record search history
+            if not SearchHistory.objects.filter(title=title, user=request.user).exists():
+                search = SearchHistory.objects.get_or_create(
+                    user = request.user,
+                    title=title,
+                    defaults={
+                        'cover_image': cover_image,
+                        'amazon_url': amazon_url,
+                        'flipkart_url': flipkart_url,
+                        'good_reads_review': good_reads_review
+                    }
+                )
+                if isinstance(search, tuple):
+                    search = search[0]
+            else:
+                search = SearchHistory.objects.get(title=title, user=request.user)
+            
+            
+            output_serializer = BookSerializer(search)
+            context = {
+                'book_data':output_serializer.data
+            }
+            return render(request, 'library/search_result.html', context)
+        
+'''This class adds the book to favorite page'''       
+class AddToFavorite(APIView):
+    permission_classes = [IsAuthenticated]
+    authentication_classes = [SessionAuthentication, BasicAuthentication]
+    
+    def post(self, request, format=None):
+        title = request.POST.get('title')
```

Comment:
when using drf, get all the values in single variable and re-use the variable.

eg : data = request.data
use data.get() to fetch all the values.